Allow InternetStreams on LAN to search for .nfo and image files for videos

snaunton opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
Allow Internet Streams on LAN to search for .nfo and image files for videos

<!--- Describe your change in detail here -->
Allow Internet Streams on LAN to search for .nfo and image files for videos

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Allow Internet Streams on LAN to search for .nfo and image files. There does not seem to be a reason to avoid this functionality. I'm sure I cannot be alone in using http to serve videos on my LAN as it is ubiquitous and fast, and clients do not need write access. For home, or any other videos which are not on scraper sites, no allow .nfo and thumb/poster images is a hindrance.

<!--- If it fixes an open issue, please link to the issue here -->
N/A

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
Have ran against my library at home which has some .nfo and images for videos which work with smb, but not http until this fix.

<!--- Include details of your testing environment, and the tests you ran to -->
Kodi on linux (Arch)
Video, .nfo and images served via http on a Synology NAS.
Added Movie and TV Show "Web server directory (HTTP)" sources 

<!--- see how your change affects other areas of the code, etc -->
It doesn't appear to.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ X] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

wsnipex committed:
[systeminfo] use timestamp for application build date from BUILDATE file,
if it exists

rikardfalkeborn committed:
[omxplayer] Make functions const correct

chewitt committed:
cmake: add freetype2 PATH_SUFFIX to FindFreeType.cmake

wsnipex committed:
[cmake] fix ninja generator when dupbuild=err

enen92 committed:
[settings] Fix regression (setting reuse)

wsnipex committed:
[ResetRenderSystem] exit early if RenderSystem hasn't been initialized

DaVukovic committed:
[settings] Hide episode thumb if unseen core wise

anxdpanic committed:
[docs] fixes for xbmcvfs.Stat

snaunton committed:
Allow InternetStreams on LAN to search for .nfo and image files

FernetMenta commented on xbmc/FileItem.cpp:
```diff
@@ -3043,7 +3043,7 @@ bool CFileItem::SkipLocalArt() const
        || StringUtils::StartsWithNoCase(m_strPath, "newplaylist://")
        || m_bIsShareOrDrive
-       || IsInternetStream()
+       || (IsInternetStream() && !URIUtils::IsOnLAN(GetPath()))
```
better use GetDynPath()

snaunton replied:
Have made change. Not sure why jenkins is failing. Builds and tests ok locally.

FernetMenta commented on xbmc/video/tags/VideoInfoTagLoaderFactory.cpp:
```diff
@@ -23,7 +24,7 @@ IVideoInfoTagLoader* CVideoInfoTagLoaderFactory::CreateLoader(const CFileItem& i
 {
   // don't try to read tags for streams
-  if (item.IsInternetStream())
+  if (item.IsInternetStream() && !URIUtils::IsOnLAN(item.GetPath()))
```
better use GetDynPath()

snaunton replied:
Have made change. Not sure why jenkins is failing. Builds and tests ok locally.

snaunton committed:
Update VideoInfoTagLoaderFactory.cpp

snaunton committed:
Update FileItem.cpp

snaunton committed:
Merge remote-tracking branch 'upstream/master'

wsnipex committed:
[cmake] make sure APP_BUILD_DATE is always set

afedchin committed:
[win32] fix wrong output size in case when system doesn't inform windowing about desktop size change in true fullscreen.

this should fix #14735

popcornmix committed:
MMALCodec: Keep videBuffer valid until next decode picture

VideoPlayerVideo expects this and it is required for DVD menus
to be updated correctly.

popcornmix committed:
MMALRenderer: Destroy pool before component

I did catch a seg-fault inside mmal_port_pool_destroy that was presumably caused by component being deleted first

Rechi commented:
@snaunton please squash your commits.

peak3d committed:
[Android] evaluate default subtitle position if GUI res != screen res

afedchin committed:
[d3d11] fixed swap chain flags on resize and re-creation to allow switching display mode.

afedchin committed:
[win32] forcible change resolution (if required) on application starts

afedchin committed:
[win10] forcible change resolution (if required) on application starts

afedchin committed:
[win32] notify about resolution change in fullscreen exclusive mode.

peak3d committed:
[Android] Reset calibrations if GUI limit changes

afedchin committed:
VideoPlayer: WinRenderBuffer - reset m_locked state on release

peak3d committed:
[Android] remove mpeg4 HD restriction / use dts for unknown pts for mpeg4

snaunton committed:
Merge remote-tracking branch 'upstream/master'

snaunton committed:
Merge branch 'master' of https://github.com/snaunton/xbmc
